src/bson.c

```diff
@@ -1060,16 +1060,6 @@ void object_to_bson(zval *object, php_phongo_bson_flags_t flags, const char *key
 				return;
 			}
 
-#if PHP_VERSION_ID >= 70000
-			tmp_ht = HASH_OF(&obj_data);
-#else
-			tmp_ht = HASH_OF(obj_data);
-#endif
-
-			if (tmp_ht && ZEND_HASH_APPLY_PROTECTION(tmp_ht)) {
-				ZEND_HASH_INC_APPLY_COUNT(tmp_ht);
-			}
-
```

Comment:
`object_to_bson()` is only called through `phongo_bson_append()`'s `IS_OBJECT` case, which is where we're now handling recursion detection and in/decrementing the hash's apply count.